packages/e2e-tests/plugins/interactive-blocks/directive-context/view.js

```diff
@@ -15,40 +37,26 @@
 			<button data-testid="async navigate" data-wp-on--click="actions.asyncNavigate">Async Navigate</button>
 		</div>`;
 
-	store( {
-		derived: {
-			renderContext: ( { context } ) => {
-				return JSON.stringify( context, undefined, 2 );
-			},
-		},
+	const { actions } = store( 'directive-context-navigate', {
 		actions: {
-			updateContext: ( { context, event } ) => {
-				const { name, value } = event.target;
-				const [ key, ...path ] = name.split( '.' ).reverse();
-				const obj = path.reduceRight( ( o, k ) => o[ k ], context );
-				obj[ key ] = value;
+			toggleText() {
+				const ctx = getContext();
+				ctx.text = "changed dynamically";
 			},
-			toggleContextText: ( { context } ) => {
-				context.text = context.text === 'Text 1' ? 'Text 2' : 'Text 1';
+			addNewText() {
+				const ctx = getContext();
+				ctx.newText = 'some new text';
 			},
-			toggleText: ( { context } ) => {
-				context.text = "changed dynamically";
-			},
-			addNewText: ( { context } ) => {
-				context.newText = 'some new text';
-			},
-			navigate: () => {
-				navigate( window.location, {
+			navigate() {
+				return navigate( window.location, {
 					force: true,
 					html,
 				} );
 			},
```

Comment:
Why returning a promise instead of using an async action?

```suggestion
			*navigate() {
				yield navigate( window.location, {
					force: true,
					html,
				} );
			},
```

Reply:
I don't remember correctly, but I think that action was meant to be called synchronously, even though it returns a promise. That's why the `asyncNavigate` exists (right below this one).

Reply:
You can call an async function synchronously, and you'll get a promise 🙂

Anyway, don't worry, it's not important.